### DISCUSSION TITLE
Generating second order elements with Triangle

### DISCUSSION THREAD

**Discussion by TheBEllis** [libigl/libigl#2154] [Q&A] [answered]
Hi libIGL team,
### Problem Statement
Can you use triangle to generate tri elements from some initial **second order** boundaries. 

**Some more explanation:**
I am using triangle to generate tri elements inside of a closed boundary. I understand that as part of triangles CLI switches you can use "-o2" to generate subparametric second order elements. However, in my current problem, I am trying to start from second order 3 node edge elements. I was wondering if there is any functionality implemented for this. I see in the comments in the header file for triangulate it says that the input element connectivity matrix should be n*2, which indicates that this might not be possible. But I just wanted to check:) Many thanks in advance!

**Edit:**
After an initial senior moment, I'm guessing it probably can't, due to the "subparametric" nature of the second order elements generated in triangle. My initial boundaries may are not in all cases subparametric.

**Comment by alecjacobson** [comment, marked as answer]
No, this is not exposed by our current wrappers. You could take a look at `igl::triangle::triangulate`. I'm guessing this wouldn't be too hard to edit to capture those. (Happy to take a PR if you figure it out and want to contribute it back).

**Reply by TheBEllis** [reply]
Thanks I will take a look and let you know if I get anywhere:)
